### PR TITLE
Update Github issues template

### PR DESCRIPTION
**Pull request by manskx**
## What does this PR do?

Update Github issues template to contain app details.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -46,14 +46,17 @@ python collect_env_details.py
 You can also fill out the list below manually.
 -->
 
+- Lightning module (e.g. Trainer, App, Work, Flow):
```
Comment (rohitgr7) [inline .github/ISSUE_TEMPLATE/bug_report.md:49]:
```suggestion
- Lightning Component (e.g. Trainer, LightningModule, App, Work, Flow):
```

**Review by rohitgr7 — APPROVED** [review]

File: requirements/collect_env_details.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
 import tqdm
 
 sys.path += [os.path.abspath(".."), os.path.abspath("")]
+import lightning_app  # noqa: E402
```
Comment (carmocca) [inline requirements/collect_env_details.py:28]: This is done in https://github.com/Lightning-AI/lightning/pull/13815 already

File: requirements/collect_env_details.py [hunk 3]
```diff
@@ -56,6 +57,7 @@ def info_packages():
         "pyTorch_version": torch.__version__,
         "pyTorch_debug": torch.version.debug,
         "pytorch-lightning": pytorch_lightning.__version__,
+        "lightning-app": lightning_app.__version__,
```
Comment (kaushikb11) [inline requirements/collect_env_details.py:60]: Some users won't have it installed, right?

**Comment by rohitgr7** [comment]
also, do you think a bug_report template should be added for app?

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (tchaton) [reply .github/ISSUE_TEMPLATE/bug_report.md:49]: I would prefer:

Lightning Abstractions (e.g. Trainer, LightningModule, LightningApp, LightningWork, LighntingFlow):

File: requirements/collect_env_details.py (hunk 3 above)
Comment (tchaton) [reply requirements/collect_env_details.py:60]: This comes with lightning

File: requirements/collect_env_details.py (hunk 2 above)
Comment (tchaton) [inline requirements/collect_env_details.py:28]:
```suggestion
from lightning import app  # noqa: E402
```

Comment (manskx) [reply requirements/collect_env_details.py:28]: nice !

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**